crates/workspace/src/which_key_layer.rs

```diff
@@ -0,0 +1,531 @@
+use crate::Workspace;
```

Comment:
@Cretezy If possible I'd like to remove any changes from the workspace crate (which is already a bit of a jumble).

Is it possible to make the WhichKeyLayer a modal? In that case we could show it only if there is no other modal showing, to avoid this:

<img width="1824" height="823" alt="Screenshot 2025-09-02 at 21 58 25" src="https://github.com/user-attachments/assets/986879af-5d71-4504-8f92-694f7910c505" />

The only thing that might cause problems is that the modal layer currently renders a bunch of space above the modal, but we can either add an item to the trait that disables that, or account for it in the positioning logic.

Reply:
Yes will take a stab at this

Reply:
How did you trigger that @ConradIrwin?